src/widget/form/settings/generalform.cpp

```diff
@@ -210,6 +211,16 @@ void GeneralForm::onTranslationUpdated()
 {
     Settings::getInstance().setTranslation(locales[bodyUI->transComboBox->currentIndex()]);
     Widget::getInstance()->setTranslation();
+    Settings::getInstance().save();
+    QMessageBox::StandardButton reply;
+      reply = QMessageBox::question(0,
+                                    tr("Do you want to restart now?"),
+                                    tr("You must qTox restart to change the language"),
+                                    QMessageBox::Yes|QMessageBox::No);
+      if (reply == QMessageBox::Yes) {
+          QProcess::startDetached(QApplication::applicationFilePath());
+          exit(12);
```

Comment:
It doesn't restart with any other exit code, just closes and doesn't start again.